### PR TITLE
[MergedRoomCreationItem] DM: RoomDescriptionText use the roomName instead of memberName

### PR DESCRIPTION
**Pull request by Claire1817**
## Type of change

- [ ] Feature
- [X] Bugfix
- [ ] Technical
- [ ] Other :

## Motivation and context
#5570 

## Screenshots / GIFs
Screenshot_1649246922

## Tests
- Create a DM 
-  Update the room name
- Check the Room creation item

## Tested devices

- [ ] Physical
- [X] Emulator
- OS version(s):

## Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [X] Pull request is based on the develop branch
- [X] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [X] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/home/room/detail/timeline/item/MergedRoomCreationItem.kt [hunk 1]
```diff
@@ -133,7 +133,10 @@ abstract class MergedRoomCreationItem : BasedMergedItem<MergedRoomCreationItem.H
         val membersCount = roomSummary?.otherMemberIds?.size ?: 0
 
         if (isDirect) {
-            holder.roomDescriptionText.text = holder.view.resources.getString(R.string.this_is_the_beginning_of_dm, roomSummary?.displayName ?: "")
+            holder.roomDescriptionText.text = holder.view.resources.getString(
+                    R.string.this_is_the_beginning_of_dm,
+                    distinctMergeData.lastOrNull()?.memberName ?: ""
```
Comment (ganfra) [inline vector/src/main/java/im/vector/app/features/home/room/detail/timeline/item/MergedRoomCreationItem.kt:138]: What if the name is changed?

Comment (Claire1817) [reply vector/src/main/java/im/vector/app/features/home/room/detail/timeline/item/MergedRoomCreationItem.kt:138]: The change only concerns the string: "This is the beginning of your direct message history with ... " 
At the end of this string we have to display the other member name and not the roomMember name. (it's the case in the web application).
About which name are you talking ? the member name or the room name ?

Comment (Claire1817) [reply vector/src/main/java/im/vector/app/features/home/room/detail/timeline/item/MergedRoomCreationItem.kt:138]: oh i understand. I think we should show the current name.

File: changelog.d/5570.bugfix [hunk 2]
```diff
@@ -0,0 +1 @@
+[MergedRoomCreationItem] DM: RoomDescriptionText use the roomName instead of memberName
```
Comment (ganfra) [inline changelog.d/5570.bugfix:1]: Use a more friendly message like:
"Use member name instead of room name in DM creation item"

**Review by ganfra — APPROVED** [review]
LGTM, thanks!

**Pull request merged**